iabgpp-encoder/src/main/java/com/iab/gpp/encoder/segment/UsNatV1CoreSegment.java

```diff
@@ -62,10 +90,143 @@ protected String encodeSegment(EncodableBitStringFields fields) {
+            "Invalid usnat sale notice / opt out combination: {" + saleOptOutNotice + " / " + saleOptOut + "}");
+      }
+    }
+
+    if (targetedAdvertisingOptOutNotice == 0) {
+      if (targetedAdvertisingOptOut != 0) {
+        throw new ValidationException("Invalid usnat targeted advertising notice / opt out combination: {"
+            + targetedAdvertisingOptOutNotice + " / " + targetedAdvertisingOptOut + "}");
+      }
+    } else if (targetedAdvertisingOptOutNotice == 1) {
+      if (saleOptOut != 1 && saleOptOut != 2) {
+        throw new ValidationException("Invalid usnat targeted advertising notice / opt out combination: {"
+            + targetedAdvertisingOptOutNotice + " / " + targetedAdvertisingOptOut + "}");
+      }
+    } else if (targetedAdvertisingOptOutNotice == 2) {
+      if (saleOptOut != 1) {
+        throw new ValidationException("Invalid usnat targeted advertising notice / opt out combination: {"
+            + targetedAdvertisingOptOutNotice + " / " + targetedAdvertisingOptOut + "}");
+      }
+    }
+
+    if (mspaServiceProviderMode == 0) {
+      if (saleOptOutNotice != 0) {
+        throw new ValidationException("Invalid usnat mspa service provider mode / sale opt out notice combination: {"
+            + mspaServiceProviderMode + " / " + saleOptOutNotice + "}");
+      }
+
+      if (sharingOptOutNotice != 0) {
+        throw new ValidationException("Invalid usnat mspa service provider mode / sharing opt out notice combination: {"
+            + mspaServiceProviderMode + " / " + sharingOptOutNotice + "}");
+      }
+
+      if (sensitiveDataLimtUserNotice != 0) {
+        throw new ValidationException(
+            "Invalid usnat mspa service provider mode / sensitive data limit use notice combination: {"
+                + mspaServiceProviderMode + " / " + sensitiveDataLimtUserNotice + "}");
+      }
+    } else if (mspaServiceProviderMode == 1) {
+      if (mspaOptOutOptionMode != 2) {
+        throw new ValidationException("Invalid usnat mspa service provider / opt out option modes combination: {"
+            + mspaServiceProviderMode + " / " + mspaServiceProviderMode + "}");
+      }
+
+      if (saleOptOutNotice != 0) {
+        throw new ValidationException("Invalid usnat mspa service provider mode / sale opt out notice combination: {"
+            + mspaServiceProviderMode + " / " + saleOptOutNotice + "}");
+      }
+
+      if (sharingOptOutNotice != 0) {
+        throw new ValidationException("Invalid usnat mspa service provider mode / sharing opt out notice combination: {"
+            + mspaServiceProviderMode + " / " + sharingOptOutNotice + "}");
+      }
+
+      if (sensitiveDataLimtUserNotice != 0) {
+        throw new ValidationException(
+            "Invalid usnat mspa service provider mode / sensitive data limit use notice combination: {"
+                + mspaServiceProviderMode + " / " + sensitiveDataLimtUserNotice + "}");
+      }
+    } else if (mspaServiceProviderMode == 2) {
```

Comment:
Hi, @iabmayank 
We have some questions regarding GPP encoding and decoding. We want to understand why there are validations for fields like mspaServiceProviderMode. We cannot find this validation in the IAB GPP documentation.
We'd appreciate your insights on this matter. Thanks in advance.